### DISCUSSION TITLE
mermaid support to add diagrams

### DISCUSSION THREAD

**Discussion by sohamakl** [typemill/typemill#443] [Ideas]
I would be delighted if it were possible in future to create diagrams directly in TypeMill, so I would like to make a suggestion for improvement. One possibility could be the support of mermaid.

Many thanks for the successful product.

**Comment by trendschau** [comment]
Thank you for your feedback! there was a mermaid plugin for Typemill Version 1, I will update it, soon.

**Comment by trendschau** [comment]
The mermaid plugin is updated to version 2 now: https://plugins.typemill.net/mermaid

**Reply by gchiesa** [reply]
Hello everyone, 

Since the plugin embeds an old version of Mermaid and I had to produce diagrams using recent features, I've created a [Pull Request](https://github.com/typemill-resources/typemill-plugins-mermaid/pull/1) to introduce the possibility to pin a specific version of mermaid via the plugin admin interface. 

The plugin is backward compatible: if no version is specified it will fallback on the embedded version. When a version is specified instead it will use unpkg CDN for obtaining the mermaid library. 

Tested successfully in https://g-c.dev with admin UI see below: 
<img width="792" alt="image" src="https://github.com/typemill/typemill/assets/4255029/6e2631bb-1800-457f-8be6-d58d0fc8d405">

**Reply by trendschau** [reply]
updated it now, I am really sorry that I missed this...

**Comment by sohamakl** [comment]
Thanks for the update.
